libp2p/host/ping.py

```diff
@@ -45,16 +66,65 @@
     return True
 
 
-async def handle_ping(stream: INetStream) -> None:
-    """``handle_ping`` responds to incoming ping requests until one side errors
-    or closes the ``stream``."""
-    peer_id = stream.muxed_conn.peer_id
+class PingService:
+    """PingService executes pings and returns RTT in miliseconds."""
 
-    while True:
+    def __init__(self, host: IHost):
+        self._host = host
+
+    async def ping(self, peer_id: PeerID) -> int:
+        stream = await self._host.new_stream(peer_id, (ID,))
         try:
-            should_continue = await _handle_ping(stream, peer_id)
-            if not should_continue:
-                return
+            rtt = await _ping(stream)
+            await _close_stream(stream)
+            return rtt
         except Exception:
-            await stream.reset()
-            return
+            await _close_stream(stream)
+            raise
+
+    async def ping_loop(
+        self, peer_id: PeerID, ping_amount: Union[int, float] = math.inf
+    ) -> "PingIterator":
+        stream = await self._host.new_stream(peer_id, (ID,))
+        ping_iterator = PingIterator(stream, ping_amount)
+        return ping_iterator
+
+
+class PingIterator:
+    def __init__(self, stream: INetStream, ping_amount: Union[int, float]):
+        self._stream = stream
+        self._ping_limit = ping_amount
```

Comment:
i think `ping_limit` is much clearer than `ping_amount`.

want to just use `ping_limit` everywhere?

Reply:
I like it! I'll change it to `ping_limit`